### PR TITLE
States -> core

### PR DESCRIPTION
**Pull request by YHerm**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/roller/RollerCommandsBuilder.java [hunk 1]
```diff
@@ -48,7 +48,11 @@ public Command rollRotations(Rotation2d rotations, double power) {
 	}
 
 	public Command stop() {
-		return new InstantCommand(roller::stop, roller).withName("Stop");
+		return new InitExecuteCommand(
```
Comment (drorta) [inline src/main/java/frc/robot/subsystems/roller/RollerCommandsBuilder.java]: runcommand?

**Review by drorta — DISMISSED** [review]

File: src/main/java/frc/robot/JoysticksBindings.java [hunk 2]
```diff
@@ -24,11 +28,36 @@ public static void configureBindings(Robot robot) {
...
+		usedJoystick.A.onTrue(robot.getSuperstructure().setState(RobotState.IDLE));
+		usedJoystick.B.onTrue(robot.getSuperstructure().setState(RobotState.PRE_SPEAKER));
+		usedJoystick.X.onTrue(robot.getSuperstructure().setState(RobotState.PRE_AMP));
+		usedJoystick.Y.onTrue(robot.getSuperstructure().setState(RobotState.SPEAKER));
+		usedJoystick.POV_UP.onTrue(robot.getSuperstructure().setState(RobotState.INTAKE));
+		usedJoystick.POV_DOWN.onTrue(robot.getSuperstructure().setState(RobotState.SHOOTER_OUTTAKE));
+		usedJoystick.POV_LEFT.onTrue(robot.getSuperstructure().setState(RobotState.AMP));
+		usedJoystick.POV_RIGHT.onTrue(robot.getSuperstructure().setState(RobotState.TRANSFER_SHOOTER_TO_ARM));
+
+		usedJoystick.L1.whileTrue(robot.getRoller().getCommandsBuilder().setPower(() -> usedJoystick.getAxisValue(Axis.LEFT_Y) * 0.7));
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/JoysticksBindings.java]: Test buttons/magic num?

File: src/main/java/frc/robot/subsystems/elbow/ElbowCommandsBuilder.java [hunk 3]
```diff
@@ -17,12 +17,12 @@ public ElbowCommandsBuilder(Elbow elbow) {
 	}
 
 	//@formatter:off
-	public Command moveToAngle(Rotation2d angle, Rotation2d tolerance) {
+	public Command moveToAngle(Rotation2d angle) {
 		return new FunctionalCommand(
 			() -> elbow.setTargetAngle(angle),
 			() -> {},
 			interrupted -> elbow.stayInPlace(),
-			() -> elbow.isAtAngle(angle, tolerance),
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/subsystems/elbow/ElbowCommandsBuilder.java:25]: If we decide to only us3 .until we should delete isFinished

File: src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package frc.robot.subsystems.flywheel;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+
+public enum FlywheelState {
+
+	DEFAULT(Rotation2d.fromRotations(0), 0.7),
+	PRE_SPEAKER(Rotation2d.fromRotations(55), 0.7);
+
+	private final Rotation2d shootingVelocity;
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java]: Just velocity will do

File: src/main/java/frc/robot/subsystems/funnel/FunnelCommandsBuilder.java [hunk 5]
```diff
@@ -17,15 +19,25 @@ public Command setPower(double power) {
...
+	public Command setPower(DoubleSupplier power) {
+		return new FunctionalCommand(
+				() -> {},
+				() -> funnel.setPower(power.getAsDouble()),
+				interrupted -> funnel.stop(),
+				() -> false,
+				funnel
+		).withName("Set power by supplier");
+	}
     //@formatter:on
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/subsystems/funnel/FunnelCommandsBuilder.java]: Put this last

File: src/main/java/frc/robot/subsystems/swerve/Swerve.java [hunk 6]
```diff
@@ -111,20 +111,22 @@ public void setHeading(Rotation2d heading) {
...
 
 	protected void resetPIDControllers() {
 		constants.xMetersPIDController().reset();
 		constants.yMetersPIDController().reset();
 		constants.rotationDegreesPIDController().reset();
 	}
 
+
 	public void saveState(SwerveState state) {
-		this.savedState = state;
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/subsystems/swerve/Swerve.java:125]: Why not this.

File: src/main/java/frc/robot/superstructure/Superstructure.java [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+	private boolean isObjectInIntake() {
+		return robot.getIntake().isObjectIn();
+	}
+
+	private boolean isObjectInFunnel() {
+		return robot.getFunnel().isObjectIn();
+	}
+
+	private boolean isReadyToTransfer() {
+		boolean isPivotReady = robot.getPivot().isAtPosition(PivotState.TRANSFER.getTargetPosition(), Tolerances.PIVOT_POSITION_TOLERANCE);
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/superstructure/Superstructure.java:69]: Why do we want a tolerance selection option?

File: src/main/java/frc/robot/superstructure/Superstructure.java [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+				new ParallelCommandGroup(
+					intakeStateHandler.setState(IntakeState.INTAKE),
+					rollerStateHandler.setState(RollerState.ROLL_IN),
+					funnelStateHandler.setState(FunnelState.STOP)
+				).until(this::isObjectInIntake),
+				new ParallelCommandGroup(
+					intakeStateHandler.setState(IntakeState.INTAKE_WITH_FUNNEL),
+					funnelStateHandler.setState(FunnelState.INTAKE),
+					rollerStateHandler.setState(RollerState.ROLL_IN)
+				).withTimeout(0.475)//.until(this::isObjectInFunnel)
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/superstructure/Superstructure.java:129]: Is the bb fixed yet? Maybe save this numbers just in case

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/superstructure/Superstructure.java (hunk 8 above)
Comment (YHerm) [reply src/main/java/frc/robot/superstructure/Superstructure.java:129]: no

File: src/main/java/frc/robot/superstructure/Superstructure.java (hunk 7 above)
Comment (YHerm) [reply src/main/java/frc/robot/superstructure/Superstructure.java:69]: if i want at score in arm less accuracy but in transfer more

File: src/main/java/frc/robot/subsystems/swerve/Swerve.java (hunk 6 above)
Comment (YHerm) [reply src/main/java/frc/robot/subsystems/swerve/Swerve.java:125]: no in constructor (that not the convention now

**Review by NoamRosenberg08 — DISMISSED** [review]

File: src/main/java/frc/robot/superstructure/Superstructure.java (hunk 8 above)
Comment (bossyshrimp68) [reply src/main/java/frc/robot/superstructure/Superstructure.java:129]: Still save that number in a const

**Pull request merged**